Public/Src/Tools/Execution.Analyzer/Analyzers.Core/XLGPlusPlus/Events.proto

```diff
@@ -22,31 +34,845 @@ message EventTypeQuery{
+    AbsolutePath_XLGpp Path = 1;
+
+    string EnumeratePatternRegex = 2;
+}
+
+message IUnsafeSandboxConfiguration_XLGpp{
+    PreserveOutputsMode_XLGpp PreserveOutputs = 1;
+
+    bool MonitorFileAccesses = 2;
+
+    bool IgnoreZwRenameFileInformation = 3;
+
+    bool IgnoreZwOtherFileInformation = 4;
+
+    bool IgnoreNonCreateFileReparsePoints = 5;
+
+    bool IgnoreSetFileInformationByHandle = 6;
+
+    bool IgnoreReparsePoints = 7;
+
+    bool IgnorePreloadedDlls = 8;
+
+    bool ExistingDirectoryProbesAsEnumerations = 9;
+
+    bool MonitorNtCreateFile = 10;
+
+    bool MonitorZwCreateOpenQueryFile = 11;
+
+    SandboxKind_XLGpp SandboxKind = 12;
+
+    bool UnexpectedFileAccessesAreErrors = 13;
+
+    bool IgnoreGetFinalPathNameByHandle = 14;
+
+    bool IgnoreDynamicWritesOnAbsentProbes = 15;
+
+    DoubleWritePolicy_XLGpp DoubleWritePolicy = 16;
+
+    bool IgnoreUndeclaredAccessesUnderSharedOpaques = 17;
+}
+
+message UnsafeOptions_XLGpp{
+    ContentHash_XLGpp PreserveOutputsSalt = 1;
 
-    // Identifies the worker which invoked this event
-    uint32 WorkerID = 2;
+    IUnsafeSandboxConfiguration_XLGpp UnsafeConfiguration = 2;
+}
+
+message StringId_XLGpp{
+    int32 Value = 1;
+}
+
+message ObservedInput_XLGpp{
+    ObservedInputType_XLGpp Type = 1;
+
+    ContentHash_XLGpp Hash = 2;
+
+    ObservedPathEntry_XLGpp PathEntry = 3;
+
+    AbsolutePath_XLGpp Path = 4;
+
+    bool IsSearchPath = 5;
+
+    bool IsDirectoryPath = 6;
+
+    bool DirectoryEnumeration = 7;
+}
+
+message ObservedPathSet_XLGpp{
+    repeated ObservedPathEntry_XLGpp Paths = 1;
+
+    repeated StringId_XLGpp ObservedAccessedFileNames = 2;
+
+    UnsafeOptions_XLGpp UnsafeOptions = 3;
+}
+
+message ProcessStrongFingerprintComputationData_XLGpp{
+    ObservedPathSet_XLGpp PathSet = 1;
+
+    ContentHash_XLGpp PathSetHash = 2;
+
+    repeated ObservedPathEntry_XLGpp PathEntries = 3;
+
+    UnsafeOptions_XLGpp UnsafeOptions = 4;
+
+    repeated StringId_XLGpp ObservedAccessedFileNames = 5;
 
-    // Identifies the source that called this event
-    string SubstSource = 3;
+    repeated StrongContentFingerPrint_XLGpp PriorStrongFingerprints = 6;
+
+    bool Succeeded = 7;
+
+    bool IsStrongFingerprintHit = 8;
+
+    StrongContentFingerPrint_XLGpp ComputedStrongFingerprint = 9;
+
+    repeated ObservedInput_XLGpp ObservedInputs = 10;
+}
+
+message DirectoryFingerprint_XLGpp{
+    ContentHash_XLGpp Hash = 1;
+}
+
+message DirectoryOutput_XLGpp{
+    DirectoryArtifact_XLGpp DirectoryArtifact = 1;
     
-    // Identifies the target of this event
-    string SubstTarget = 4;
+    repeated FileArtifact_XLGpp fileArtifactArray = 2;
+}
+
+message DirectoryArtifact_XLGpp{
+    bool IsValid = 1;
+
+    AbsolutePath_XLGpp Path = 2;
+
+    uint32 PartialSealID = 3;
+
+    bool IsSharedOpaque = 4;
+}
+
+//
+//  Event Data
+//
+
+// The FileArtifactContentDecided Event message
+message FileArtifactContentDecidedEvent_XLGpp{
+    string UUID = 99;

```

Comment:
>string UUID = 99; [](start = 4, length = 17)

so you are using this filed #'99' for uniqueness. will it actually work if you add new elements? I think there is a big difference between:
1, 2, 3, NEW_FIELD
and
1, 2, NEW_FIELD, 3

Reply:
By elements, do you mean if I add more fields to this message? No, protobuf will not care about which order I add in new fields so long as i assign it a unique number that hasn't been used before.

---
In reply to: [308897609](https://github.com/microsoft/BuildXL/pull/671#discussion_r308897609) [](ancestors = 308897609)